packages/flutter/test/widgets/framework_test.dart

```diff
@@ -739,6 +739,73 @@ void main() {
     expect(state.didChangeDependenciesCount, 3);
     expect(state.deactivatedCount, 2);
   });
+  testWidgets('StatefulElement subclass can decorate State.build', (WidgetTester tester) async {
+    bool isDidChangeDependenciesDecorated;
+    bool isBuildDecorated;
+
+    final Widget child = Decorate(
+      didChangeDependencies: (bool value) {
+        isDidChangeDependenciesDecorated = value;
+      },
+      build: (bool value) {
+        isBuildDecorated = value;
+      }
+    );
+
+    await tester.pumpWidget(Inherited(0, child: child));
+
+    expect(isBuildDecorated, isTrue);
+    expect(isDidChangeDependenciesDecorated, isFalse);
+
+    await tester.pumpWidget(Inherited(1, child: child));
+
+    expect(isBuildDecorated, isTrue);
+    expect(isDidChangeDependenciesDecorated, isFalse);
+  });
+}
+
+class Decorate extends StatefulWidget {
+  const Decorate({Key key, this.didChangeDependencies, this.build}) : super(key: key);
+
+  final void Function(bool isInBuild) didChangeDependencies;
+  final void Function(bool isInBuild) build;
+
+
+  @override
+  _DecorateState createState() => _DecorateState();
+
+  @override
+  DecorateElement createElement() => DecorateElement(this);
+}
+
+class DecorateElement extends StatefulElement {
+  DecorateElement(Decorate widget): super(widget);
+
+  bool isDecorated = false;
+
+  @override
+  Widget build() {
+    try {
+      isDecorated = true;
+      return super.build();
+    } finally {
+      isDecorated = false;
+    }
+  }
+}
+
+class _DecorateState extends State<Decorate> {
+  @override
+  void didChangeDependencies() {
+    super.didChangeDependencies();
+    widget.didChangeDependencies?.call((context as DecorateElement).isDecorated);
```

Comment:
assert non null above so this can just be `widget.didChangeDependencies((context as DecorateElement).isDecorated);`